### PR TITLE
PUBDEV-4694: Make CompressedTree smaller by storing common metatada elsewhere

### PR DESCRIPTION
**Pull request by michalkurka**
This object can later store information shared among all Trees, for
now it just holds the keys of the trees

### CODE REVIEW

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@mmalohlava please review

File: h2o-algos/src/main/java/hex/tree/CompressedForest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          trees[t][i] = DKV.get(treek[i]).get();
+    }
+    return new LocalCompressedForest(trees, _domains);
+  }
+
+  /**
+   * Node-local representation of a collection of trees.
+   * Is not meant to be Serialized/Iced or send over the wire.
+   */
+  public static class LocalCompressedForest {
```
Comment (mmalohlava) [inline h2o-algos/src/main/java/hex/tree/CompressedForest.java]: This is nice trick to avoid accidental serialization!

**Comment by mmalohlava** [comment]
QQ: what about MOJO? Any impact?

**Comment by michalkurka** [comment]
@mmalohlava there is no impact on MOJO, I only changed internal APIs. This is also fully backwards compatible, we don't need to change version of MOJO readers/writers.

**Pull request merged**